File: columbu/CATscala/src/main/java/com/columbu/db/levelDB/store/RevokingDatabase.java
package com.galileo.db.levelDB.store;

import org.tron.core.db2.common.IRevokingDB;
import com.galileo.db.levelDB.core.ISession;
import com.galileo.core.exception.RevokingStoreIllegalStateException;

public interface RevokingDatabase {

  ISession buildSession();

  ISession buildSession(boolean forceEnable);

  void add(IRevokingDB revokingDB);

  void merge() throws RevokingStoreIllegalStateException;

  void revoke() throws RevokingStoreIllegalStateException;

  void commit() throws RevokingStoreIllegalStateException;

  void pop() throws RevokingStoreIllegalStateException;

  void fastPop() throws RevokingStoreIllegalStateException;

  void enable();

  int size();

  void check();

  void setMaxSize(int maxSize);

  void disable();

  void shutdown();
}

File: columbu/CATscala/src/main/scala/com/columbu/db/levelDB/store/RevokingDatabase.scala
package com.galileo.db.levelDB.store

import com.galileo.core.exception.RevokingStoreIllegalStateException
import com.galileo.db.levelDB.common.IRevokingDB
import com.galileo.db.levelDB.core.ISession

trait RevokingDatabase {

  def buildSession: ISession

  def buildSession(forceEnable: Boolean): ISession

  def add(revokingDB: IRevokingDB): Unit

  @throws[RevokingStoreIllegalStateException]
  def merge(): Unit

  @throws[RevokingStoreIllegalStateException]
  def revoke(): Unit

  @throws[RevokingStoreIllegalStateException]
  def commit(): Unit

  @throws[RevokingStoreIllegalStateException]
  def pop(): Unit

  @throws[RevokingStoreIllegalStateException]
  def fastPop(): Unit

  def enable(): Unit

  def size: Int

  def check(): Unit

  def setMaxSize(maxSize: Int): Unit

  def disable(): Unit

  def shutdown(): Unit
}
